### PR TITLE
Feed

### PR DESCRIPTION
**Pull request by adescoteaux1**
Creating the feed endpoint. Created a feedPost struct which should have all the firelds required to display each review component for the frontend feed page based on screenshot in the issue. created a reviewStat func that will calculate upvotes/downvotes and comments for display and then add those to each review after the sql query runs

## Description

[Link to Ticket](https://github.com/GenerateNU/platnm/issues/115#issue-2616950656)

## How Has This Been Tested?

image

## Checklist:
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] Some tests are included that test my code

### CODE REVIEW

File: backend/internal/storage/postgres/schema/user.go [hunk 1]
```diff
@@ -175,6 +176,106 @@ func (r *UserRepository) GetUserProfile(ctx context.Context, id uuid.UUID) (*mod
...
+		r.rating, 
+		r.created_at, 
+		r.updated_at,
+		COALESCE(a.cover, t.cover) AS media_cover, 
+		COALESCE(a.title, t.title) AS media_title, 
+		COALESCE(a.artists, t.artists) AS media_artist
+	FROM review r
+	INNER JOIN follower f ON f.followee_id = r.user_id
+	INNER JOIN "user" u ON u.id = r.user_id
+  LEFT JOIN (
```
Comment (ddusichka) [inline backend/internal/storage/postgres/schema/user.go:207]: this is some gnarly sql omg but looks solid to me

File: backend/internal/models/feedPost.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+type FeedPost struct {
+	ID             int       `json:"id"`
+	UserID         string    `json:"user_id"`
+	Username       string    `json:"username"`
+	ProfilePicture string    `json:"profile_picture"`
+	MediaType      MediaType `json:"media_type"`
+	MediaID        int       `json:"media_id"`
+	Rating         int       `json:"rating"`
```
Comment (ddusichka) [inline backend/internal/models/feedPost.go:14]: can we add Comment too? representing the actual text associated with the review? this is potentially null

File: backend/internal/models/feedPost.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	Comments       int       `json:"comments"`
```
Comment (ddusichka) [inline backend/internal/models/feedPost.go:15]: can we rename this to CommentCount just for clarity

**Review by ddusichka — COMMENTED** [review]
GREAT JOB this is def the trickiest endpoint we have sql-wise so far

once comment is added I think this is all set!

File: backend/internal/models/feedPost.go (hunk 2 above)
Comment (adescoteaux1) [reply backend/internal/models/feedPost.go:14]: Oh yes, totally missed that

**Review by ddusichka — APPROVED** [review]
yay! now we can make a FE feed ticket 😆

**Pull request merged**